ocaml/xapi/xapi_host.ml

```diff
@@ -596,10 +596,24 @@ let list_methods ~__context =
 let is_slave ~__context ~host = not (Pool_role.is_master ())
 
 let ask_host_if_it_is_a_slave ~__context ~host =
-  let local_fn = is_slave ~host in
-  Server_helpers.exec_with_subtask ~__context "host.ask_host_if_it_is_a_slave" (fun ~__context ->
-    (Message_forwarding.do_op_on_localsession_nolivecheck ~local_fn ~__context
-      ~host (fun session_id rpc -> Client.Client.Pool.is_slave rpc session_id host)))
+  let ask_and_warn_when_slow ~__context =
+    let local_fn = is_slave ~host in
+    let timeout = 10 in
+    let task_name = Context.get_task_id __context |> Ref.string_of in
+    let ip, uuid = Db.Host.get_address ~__context ~self:host, Db.Host.get_uuid ~__context ~self:host in
+    let rec log_host_slow_to_respond timeout' () =
+      D.warn "ask_host_if_it_is_a_slave: host taking a long time to respond - IP: %s; uuid: %s" ip uuid;
```

Comment:
It's fine to include the name of the function; another option is to use `__LOC__` which contains the current source code location.

Reply:
that's cool, i didn't know that's what `__LOC__` was for

Reply:
I will stick with using the function name unless you have an objection